docs/src/references/c8y-configuration-management.md

```diff
@@ -274,107 +274,125 @@ The HTTP filetransfer feature of the `tedge_agent` provides the service to trans
     Payload: ` <temporary URL> `
        
     TODO: Investigate and decide about topic structure and payload in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307). Unhappy paths also to be considered here.
   
-  * C8Y config plugin: recognizes the MQTT notification about the uploaded file and the temprary-URL, 
-                       and uses the filetransfers local API to obtain the file from the _temprary URL_ to some plugin specific local filesystem path (e.g. `/tmp/c8y-cfg-plugin/<childid>_<cfg-type>`.
-
-    * TODO: Investigate and decide about filetransfer's local API in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307), according to topics below:
- 
-      * local API must avoid that there are two copies of the same file at any point in time
-      * local API must avoid that the file content is transfered/copied from one to another place, when the plugin _obtains_ the file. 
-        Instead just the directory link may change (as it is the case for bash `mv` command).
-      * file is removed from filetransfer's specific folder when the file was successfully obtained to the plugin's path.
-      
-      Reasons for requirements above: Saving disc-space and CPU resources for large files.
-
-    * TO-BE-DEFINED:
-    
-      * Decide which error paths to be considered.
-      * Decide if any timeout to be considerd when waiting for upload notification from external device.
+  6) C8Y config plugin: recognizes the MQTT notification about the uploaded file and the _temprary URL_, 
+                       and downloads the file using HTTP the filetransfer with the _temporary URL_ to some temporary location.
+  
+    Example for HTTP GET request: `http://<ip address of thin-edge device>/<temporary URL>`
 
-  * C8Y config plugin: sends the obtained file to C8Y
+  7) C8Y config plugin: sends the downloaded file to C8Y
 
-  * C8Y config plugin: removed the file form local filesystem
+  8) C8Y config plugin: removes the downloaded file from temporary location
   
 
 ## Pushing configuration file update to child device from cloud
 
-  A configuration file snapshot is pushed to an external device as outlined below:
+A configuration file snapshot is pushed to an external device as illustrated below.
 
-  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin<br/>
-    Format of C8Y SmartREST message for config send operation: `524,<childid>,<URL>,<config type>`. See [C8Y SmartREST doc](https://cumulocity.com/guides/reference/smartrest-two/#download-configuration-file-with-type-524)<br/>
-    Example: `524,child1,http://www.my.url,bar.conf`
-  * C8Y config plugin: downloads the file based on the URL received from C8Y, to some temporary location
+```mermaid
+sequenceDiagram
+    participant External Device
+    participant Tedge Agent
+    participant C8Y Cfg Plugin
+    participant C8Y Cloud
+    participant WebSpace
```

Comment:
```suggestion
    participant WebSpace
```
Do we really need this participant? I'm assuming that you added it to highlight the fact that the config file could exist anywhere on the web and not necessarily in C8Y cloud itself. I'd say that it can be simplified to C8Y cloud to make the diagram simpler.

Reply:
That's good simplification sugesstion. I will remove that WebSpace thing and use C8Y instead.

Reply:
Done